src/client/Microsoft.Identity.Client/AppConfig/ApplicationConfiguration.cs

```diff
@@ -23,9 +24,18 @@ namespace Microsoft.Identity.Client
 {
     internal sealed class ApplicationConfiguration : IAppConfig
     {
-        public ApplicationConfiguration(bool isConfidentialClient) 
+        public ApplicationConfiguration(ApplicationType applicationType) 

```

Comment:
I think `IsManagedIdentity` and `IsConfidentialClient` are not needed and should be replaced with `ApplicationType` property.

If that results in many changes, then maybe refactor in a follow up PR, not this one.

Reply:
It's a big PR already, just bite the bullet imo. If changes are in a single commit, it's easy to review (or to skip review)

Reply:
I will create another PR for these changes, thanks for the suggestion @pmaytak

Reply:
https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/4113